### PR TITLE
The HW JS #2 with the tasks 1/2/3.

### PR DESCRIPTION
**Pull request by Labrosimova**
Tasks are ready for review. Thank you for your work and lessons!

### CODE REVIEW

File: JS_HW_2/HW_JS_2_task_1.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const minAge = 18; // 1. Создать переменную "minAge" и присвоить ей значение 18
+const maxAge = 60; //   2. Создать переменную "maxAge" и присвоить ей значение 60
+let age; //3. Создать переменную "age", в которую вы будете вводить возраст при выполнении программы
+age = 61; // 5. Проверить задание со следующими значениями в переменной age: 10, 17, 18, 19, 59, 60, 61
```
Comment (VladimirLQA) [inline JS_HW_2/HW_JS_2_task_1.js]:
```suggestion
let age = 61; //3. Создать переменную "age", в которую вы будете вводить возраст при выполнении программы
 // 5. Проверить задание со следующими значениями в переменной age: 10, 17, 18, 19, 59, 60, 61
```

File: JS_HW_2/HW_JS_2_task_2.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+//     default:
+//     console.log("Technical work");
+
+// };
+// 3. Преобразовать Task 2 - 1 таким образом, чтобы значение НАПРИМЕР '2' (т.е. ЛЮБАЯ строка в которой лежат ТОЛЬКО ЦИФРЫ) пропускалось, преобразовываясь в number
+const minAge = 18; 
+const maxAge = 60;
+let age;
+age = "18";
```
Comment (VladimirLQA) [inline JS_HW_2/HW_JS_2_task_2.js]:
```suggestion
let age = "18"; 
```

File: JS_HW_2/HW_JS_2_task_2.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+const minAge = 18; 
+const maxAge = 60;
+let age;
+age = "18"; 
+const numberedAge = age*1; // 3. Преобразовать Task 2 - 1 таким образом, чтобы значение НАПРИМЕР '2' (т.е. ЛЮБАЯ строка в которой лежат ТОЛЬКО ЦИФРЫ) пропускалось, преобразовываясь в number
+switch (true) {
+    case typeof (numberedAge) !== "number":
+    console.log("Oops! The value is NaN. Please try again!");
+    break;
+
```
Comment (VladimirLQA) [inline JS_HW_2/HW_JS_2_task_2.js]: Follow the same logic as in the 3-rd task, use `isNaN` right away

File: JS_HW_2/HW_JS_2_task_3.html [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    <title>Document</title>
+</head>
+<body>
+    <script>
+        // Преобразовать Task 2* таким образом, чтобы age принимался через prompt в привязанной вёрстке, а результат выводился в alert
+"use strict";
+const minAge = 18; 
+const maxAge = 60;
+let age;
+age = prompt("What's your age?",);
```
Comment (VladimirLQA) [inline JS_HW_2/HW_JS_2_task_3.html]:
```suggestion
let age = prompt("What's your age?"); 
```

File: JS_HW_2/HW_JS_2_task_3.html [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+<body>
+    <script>
+        // Преобразовать Task 2* таким образом, чтобы age принимался через prompt в привязанной вёрстке, а результат выводился в alert
+"use strict";
+const minAge = 18; 
+const maxAge = 60;
+let age;
+age = prompt("What's your age?",); 
+const numberedAge = Number(age); 
+alert(numberedAge);
```
Comment (VladimirLQA) [inline JS_HW_2/HW_JS_2_task_3.html]:
```suggestion
```

File: JS_HW_2/HW_JS_2_task_3.html [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+if (isNaN(numberedAge)) {
+    alert("Oops! The value is NaN. Please try again!");
+} else {
+switch (true) {
+    case typeof (numberedAge) !== "number":
+    alert("Oops! The value is NaN. Please try again!");
+    break;
```
Comment (VladimirLQA) [inline JS_HW_2/HW_JS_2_task_3.html]:
```suggestion
```
Excess check, you've already converted it to a number

File: JS_HW_2/HW_JS_2_task_3.html [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+</head>
+<body>
+    <script>
+        // Преобразовать Task 2* таким образом, чтобы age принимался через prompt в привязанной вёрстке, а результат выводился в alert
+"use strict";
+const minAge = 18; 
+const maxAge = 60;
+let age;
+age = prompt("What's your age?",); 
+const numberedAge = Number(age);
```
Comment (VladimirLQA) [inline JS_HW_2/HW_JS_2_task_3.html]:
```suggestion
const numberedAge = Number(age); 
```
Move this line to `else` block after `isNaN` check

**Review by VladimirLQA — APPROVED** [review]

File: JS_HW_2/HW_JS_2_task_3.html (hunk 7 above)
Comment (Labrosimova) [reply JS_HW_2/HW_JS_2_task_3.html]: @VladimirLQA 
Thank you for reviewing.
I don't understand; please, can you explain me, where am I wrong? 
According to the last comment: I declared the variable (const numberedAge = Number(age);) before the *if* block for checking: if the converted to number type value is not equal to NaN. But if I move this variable to *else* block, the check in the *if* block doesn't know about the -> const numberedAge = Number(age);

Comment (VladimirLQA) [reply JS_HW_2/HW_JS_2_task_3.html]: Firstly, you need to check value for isNaN then convert it, if the check passed
With this approach, we do not perform an unnecessary value conversion action
```js
if (isNaN(age)) {
    alert("Oops! The value is NaN. Please try again!");
} else {
const numberedAge = Number(age);
switch (true) {
```

**Pull request merged**